src/world.rs

```diff
@@ -878,6 +881,18 @@ impl World {
         ColumnMut::new(entities, archetypes, PhantomData)
     }
 
+    /// Despawn `entity`, yielding a [`DynamicBundle`] of its components
+    ///
+    /// Useful for moving entities between worlds.
+    pub fn take(&mut self, entity: Entity) -> Result<TakenEntity<'_>, NoSuchEntity> {
+        let loc = self.entities.free(entity)?;
```

Comment:
`World::despawn` calls `World::flush` before doing anything. Is that unnecessary here?

Reply:
Good catch; absolutely necessary.